### ISSUE TITLE
Cannot use numpy.int32 or numpy.int64 as indices to access single array element 

### ISSUE THREAD

**Issue by piyueh** [nv-legate/legate.numpy#13]
## Problem

When using `numpy.int32` or `numpy.int64` as indices to get an element from a 2D array, the code triggers something that is not implemented, i.e., `NotImplementedError`. However, if converting the indices to native `int`, then everything works.

I'm not sure if this is just an unimplemented feature, or if something's wrong, as this should be basic indexing and no advanced indexing involved.

`numpy.int64` seems to be the default type of the element returned by looping a NumPy integer array. So I feel this is a common use case.

## To reproduce

1. step 1: create two python scripts: `test_1.py` and `test_2.py`
   - In `test_1.py`
     ```python
     from legate import numpy
     a = numpy.random.random((1000, 2000))
     idx = numpy.random.randint(0, 99, 10, int)
     idy = numpy.random.randint(0, 99, 10, int)

     for i, j in zip(idx, idy):
         print("index type: ({}, {}); ".format(type(i), type(j)), end="")
         print("a[i, j] = {}".format(a[i, j]))
     ```
   - In `test_2.py`
     ```python
     from legate import numpy
     a = numpy.random.random((1000, 2000))
     idx = numpy.random.randint(0, 99, 10, int)
     idy = numpy.random.randint(0, 99, 10, int)

     for i, j in zip(idx, idy):
         i, j = int(i), int(j)
         print("index type: ({}, {}); ".format(type(i), type(j)), end="")
         print("a[i, j] = {}".format(a[i, j]))
     ```
2. step 2: run both scripts
   ```shell
   $ legate --cpus 1 ./test_1.py
   ```
   and
   ```python
   $ legate --cpus 1 ./test_2.py

## Expected behavior
Both `test_1.py` and `test_2.py` should both output 10 lines of message in a format of `index type: (class XXX, class XXX); a[i, j] = YYYYYYYYYYY`. `XXX` is either `numpy.int64` or `int`, depending on it's `test_1.py` or `test_2.py`, while `YYYYYYYYYYY` are random numbers.

## Actual output
`test_1.py` reports an error (but successfully prints the first part on the message of the first line, i.e, `index type: ...`):
```python
Traceback (most recent call last):
  File "<prefix>/lib/python3.8/site-packages/legion_top.py", line 394, in legion_python_main
    run_path(args[start], run_name='__main__')
  File "<prefix>/lib/python3.8/site-packages/legion_top.py", line 193, in run_path
    exec(code, module.__dict__, module.__dict__)
  File "./test_1.py", line 8, in <module>
    print("a[i, j] = {}".format(a[i, j]))
  File "<prefix>/lib/python3.8/site-packages/legate/numpy/array.py", line 381, in __getitem__
    shape=None, thunk=self._thunk.get_item(key, stacklevel=2)
  File "<prefix>/lib/python3.8/site-packages/legate/numpy/deferred.py", line 343, in get_item
    index_array = self._create_indexing_array(
  File "<prefix>/lib/python3.8/site-packages/legate/numpy/deferred.py", line 332, in _create_indexing_array
    raise NotImplementedError("need support for concatenating arrays")
NotImplementedError: need support for concatenating arrays
index type: (<class 'numpy.int64'>, <class 'numpy.int64'>);
```

**Comment by lightsighter** [comment]
Should be fixed with `c6256489e399`

**Comment by piyueh** [comment]
I confirm this is working. Thanks! Closing the issue.
